internal/dcache/cluster_manager/cluster_manager_impl.go

```diff
@@ -339,12 +365,141 @@ func (cmi *ClusterManagerImpl) punchHeartBeat(rvList []dcache.RawVolume) {
 	}
 }
 
+func (cmi *ClusterManagerImpl) updateStorageClusterMapIfRequired() {
+	clusterMapBytes, etag, err := getClusterMap()
+	if err != nil {
+		log.Err("updateStorageClusterMapIfRequired: GetClusterMap failed. err %v", err)
+		return
+	}
+	var clusterMap dcache.ClusterMap
+	if err := json.Unmarshal(clusterMapBytes, &clusterMap); err != nil {
+		log.Err("updateStorageClusterMapIfRequired: failed to unmarshal clusterMapBytes, error: %v", err)
+		return
+	}
+
+	now := time.Now().Unix()
+	if clusterMap.LastUpdatedAt > now {
+		log.Warn("updateStorageClusterMapIfRequired: skipping as it is already updated LastUpdatedAt(%s) and current time(%s) ", clusterMap.LastUpdatedAt, now)
+		return
+	}
+	clusterMapAge := now - clusterMap.LastUpdatedAt
+
+	const thresholdEpochTime = 60
+	stale := clusterMapAge > int64(clusterMap.Config.ClustermapEpoch+thresholdEpochTime)
+	leader := clusterMap.LastUpdatedBy == cmi.nodeId
+
+	// Skip if we're neither leader nor the map is stale
+	if !leader && !stale {
+		log.Info("updateStorageClusterMapIfRequired: skipping, node (%s) is not leader and Cluster map is fresh (last update %d).", clusterMap.LastUpdatedAt, cmi.nodeId)
+		return
+	} else {
+		// TODO{Akku}: UpdateClusterMapStart should also expect a nodeId. In case leader is dead
```

Comment:
why?
Caller should pass updated clusterMapBytes with state set to "checking" and last_updated_by set to this node's nodeid.